scipy/spatial/transform/tests/test_rigid_transform.py

```diff
@@ -228,103 +302,126 @@ def test_from_exp_coords():
+    xp_assert_close(
         RigidTransform.from_dual_quat(
-            np.array([0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0])).as_matrix(),
-        np.eye(4), atol=1e-12)
-    assert_allclose(
+            xp.asarray([0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0])).as_matrix(),
+        xp.eye(4), atol=1e-12)
+    xp_assert_close(
         RigidTransform.from_dual_quat(
-            np.array([1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]),
+            xp.asarray([1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]),
             scalar_first=True).as_matrix(),
-        np.eye(4), atol=1e-12)
+        xp.eye(4), atol=1e-12)
 
     # only translation
     actual = RigidTransform.from_dual_quat(
-        [0, 0, 0, 1, 0.25, 0.15, -0.7, 0])
-    expected_matrix = np.array([
+        xp.asarray([0, 0, 0, 1, 0.25, 0.15, -0.7, 0]))
+    expected_matrix = xp.asarray([
         [1, 0, 0, 0.5],
         [0, 1, 0, 0.3],
         [0, 0, 1, -1.4],
         [0, 0, 0, 1]
     ])
-    assert_allclose(actual.as_matrix(), expected_matrix, atol=1e-12)
+    xp_assert_close(actual.as_matrix(), expected_matrix, atol=1e-12)
     actual = RigidTransform.from_dual_quat(
-        [1, 0, 0, 0, 0, 0.25, 0.15, -0.7], scalar_first=True)
-    expected_matrix = np.array([
+        xp.asarray([1, 0, 0, 0, 0, 0.25, 0.15, -0.7]), scalar_first=True)
+    expected_matrix = xp.asarray([
         [1, 0, 0, 0.5],
         [0, 1, 0, 0.3],
         [0, 0, 1, -1.4],
         [0, 0, 0, 1]
     ])
-    assert_allclose(actual.as_matrix(), expected_matrix, atol=1e-12)
+    xp_assert_close(actual.as_matrix(), expected_matrix, atol=1e-12)
 
     # only rotation
-    actual_rot = Rotation.from_euler("xyz", [65, -13, 90], degrees=True)
+    actual_rot = Rotation.from_euler("xyz", xp.asarray([65, -13, 90]), degrees=True)
     actual = RigidTransform.from_dual_quat(
-        np.hstack((actual_rot.as_quat(), np.zeros(4))))
-    expected_matrix = np.eye(4)
-    expected_matrix[:3, :3] = actual_rot.as_matrix()
-    assert_allclose(actual.as_matrix(), expected_matrix, atol=1e-12)
+        xp.concat((actual_rot.as_quat(), xp.zeros(4)), axis=-1))
+    expected_matrix = xp.eye(4)
+    expected_matrix = xpx.at(expected_matrix)[..., :3, :3].set(actual_rot.as_matrix())
+    xp_assert_close(actual.as_matrix(), expected_matrix, atol=1e-12)
 
     actual = RigidTransform.from_dual_quat(
-        np.hstack((actual_rot.as_quat(scalar_first=True), np.zeros(4))),
+        xp.concat((actual_rot.as_quat(scalar_first=True), xp.zeros(4)), axis=-1),
         scalar_first=True)
-    expected_matrix = np.eye(4)
-    expected_matrix[:3, :3] = actual_rot.as_matrix()
-    assert_allclose(actual.as_matrix(), expected_matrix, atol=1e-12)
+    expected_matrix = xp.eye(4)
+    expected_matrix = xpx.at(expected_matrix)[..., :3, :3].set(actual_rot.as_matrix())
+    xp_assert_close(actual.as_matrix(), expected_matrix, atol=1e-12)
 
     # rotation and translation
+    # The tolerance is set to 1e-8, because xp_assert_close compares the absolute
```

Comment:
Following up from the comments on the other PR, I think the tolerance change here is fine and doesn't need to be broken out to another PR. However I do believe that there is still a default rtol applied for floating point numbers, so this comment should change